lib/spack/spack/build_environment.py

```diff
@@ -336,6 +337,10 @@ def set_build_environment_variables(pkg, env, dirty=False):
     env.set(SPACK_SHORT_SPEC, pkg.spec.short_spec)
     env.set(SPACK_DEBUG_LOG_DIR, spack.spack_working_dir)
 
+    # Find ccache binary and hand it to build environment
+    if spack.ccache:
+        env.set(SPACK_CCACHE_BINARY, 'ccache')
```

Comment:
Here is the crux, how would I detect a `ccache` binary built by spack previously.